test-scripts/check.sh

```diff
@@ -144,6 +144,62 @@ test_extract_different_works() {
   verify_snapshot "$pack":"$snapshot_b"
 }
 
+test_extract_file_modes_preserved() {
+  umask 0022
+
+  touch foobar
+  [[ "$(stat -c %A foobar)" == "-rw-r--r--" ]] || {
+    echo "Failed."
+    exit 1
+  }
+
+  # Pack an empty file called foobar, change the mode.
+  "$elfshaker" store s0
+  chmod +x foobar
+  "$elfshaker" store s1
+  chmod -x foobar
+  "$elfshaker" store s2
+
+  # Verify that extract from loose snapshot sets the expected mode.
+  "$elfshaker" extract s0
+  [[ "$(stat -c %A foobar)" == "-rw-r--r--" ]] || {
+    echo "Failed."
+    exit 1
+  }
+  "$elfshaker" extract s1
+  [[ "$(stat -c %A foobar)" == "-rwxr-xr-x" ]] || {
+    echo "Failed."
+    exit 1
+  }
+  "$elfshaker" extract s2
+  [[ "$(stat -c %A foobar)" == "-rw-r--r--" ]] || {
+    echo "Failed."
+    exit 1
+  }
+
+  # Verify that extracting from a pack  sets the mode.
+  "$elfshaker" pack p0
+  "$elfshaker" gc --loose-snapshots
+
+  "$elfshaker" extract p0:s0
+  [[ "$(stat -c %A foobar)" == "-rw-r--r--" ]] || {
+    echo "Failed."
+    exit 1
+  }
+  "$elfshaker" extract p0:s1
+  [[ "$(stat -c %A foobar)" == "-rwxr-xr-x" ]] || {
+    echo "Failed."
+    exit 1
+  }
+  "$elfshaker" extract p0:s2
+  [[ "$(stat -c %A foobar)" == "-rw-r--r--" ]] || {
+    echo "Failed."
+    exit 1
+  }
+
+  rm elfshaker_data/packs/p0.pack{,.idx}
```

Comment:
This is fine, we should probably update `before_test` to clean up test packs automatically in the future.

Reply:
I'll leave this as is for now. I can see that `before_test` currently specifically excludes the packs directory and I don't want to dig into this for now.